### PR TITLE
Adding DST on timezone names

### PR DESCRIPTION
**Pull request by Skrypt**
re-fixes completely #3510 

image

### CODE REVIEW

File: src/OrchardCore/OrchardCore/Modules/Services/Clock.cs [hunk 1]
```diff
@@ -76,8 +77,9 @@ private ITimeZone CreateTimeZone(DateTimeZone dateTimeZone)
             }
 
             var offset = dateTimeZone.GetZoneInterval(CurrentInstant).StandardOffset;
+            var dstOffset = dateTimeZone.GetZoneInterval(CurrentInstant).WallOffset;
```
[the remark this answers is not in the document]
Comment (Skrypt) [reply src/OrchardCore/OrchardCore/Modules/Services/Clock.cs]: right

File: src/OrchardCore/OrchardCore/Modules/Services/Clock.cs [hunk 2]
```diff
@@ -28,8 +28,9 @@ public ITimeZone[] GetTimeZones()
                 let zoneId = location.ZoneId
                 let tz = DateTimeZoneProviders.Tzdb[zoneId]
                 let offset = tz.GetZoneInterval(CurrentInstant).StandardOffset
+                let dst = tz.GetZoneInterval(CurrentInstant).WallOffset
```
[the remark this answers is not in the document]
Comment (Skrypt) [reply src/OrchardCore/OrchardCore/Modules/Services/Clock.cs]: right

**Comment by Skrypt** [comment]
Let me check if I can't just reused also the DateTimeZone simply.

**Comment by sebastienros** [comment]
That too ;) Just ensure it doesn't add any extra dependency

**Comment by Skrypt** [comment]
The OrderBy thing is really strange because we already OrderBy(StandardOffset, zoneId) in the GetTimeZones() method. I should never need to do Clock.GetTimeZones().OrderBy(t => t.ToString()) on the view.

**Comment by Skrypt** [comment]
I removed it locally and the OrderBy works from the GetTimeZones() method now. Well, lets keep it in the View too to be sure.

**Pull request merged**